pkgs/development/compilers/julia/patches/1.10/0001-hardcode-macos-SDK.patch

```diff
@@ -0,0 +1,27 @@
+From d141939b3d6c51f5b4e166ba3a0c569e31e65321 Mon Sep 17 00:00:00 2001
+From: MilesCranmer <miles.cranmer@gmail.com>
+Date: Fri, 7 Jun 2024 03:04:20 +0100
+Subject: [PATCH] hardcode macos SDK
+
+---
+ base/Makefile | 4 ++--
+ 1 file changed, 2 insertions(+), 2 deletions(-)
+
+diff --git a/base/Makefile b/base/Makefile
+index ad2bb6a63c..666eefac36 100644
+--- a/base/Makefile
++++ b/base/Makefile
+@@ -89,8 +89,8 @@ else
+ 	@echo "const DARWIN_FRAMEWORK = false" >> $@
+ endif
+ ifeq ($(OS), Darwin)
+-	@echo "const MACOS_PRODUCT_VERSION = \"$(shell sw_vers -productVersion)\"" >> $@
+-	@echo "const MACOS_PLATFORM_VERSION = \"$(shell xcrun --show-sdk-version)\"" >> $@
++	@echo "const MACOS_PRODUCT_VERSION = \"14.1.2\"" >> $@
```

Comment:
Can we derive these from the Nix derivations for the SDK somehow, rather than hardcode them? Maybe rather than provide a .patch file, we could run `sed` commands at patch time.

Reply:
What's the right way to do this? Should I run a `sed` command right inside `postPatch`? I didn't know why `shell sw_vers` didn't work in the Makefile so wasn't sure where `sw_vers` would actually work.

Reply:
This is what LLVM build has in `default.nix`:

```nix
    # As of LLVM 15, marked as XFAIL on arm64 macOS but lit doesn't seem to pick
    # this up: https://github.com/llvm/llvm-project/blob/c344d97a125b18f8fed0a64aace73c49a870e079/llvm/test/MC/ELF/cfi-version.ll#L7
    rm test/MC/ELF/cfi-version.ll

    # This test tries to call `sw_vers` by absolute path (`/usr/bin/sw_vers`)
    # and thus fails under the sandbox:
  '' + (if lib.versionAtLeast release_version "16" then ''
    substituteInPlace unittests/TargetParser/Host.cpp \
      --replace '/usr/bin/sw_vers' "${(builtins.toString darwin.DarwinTools) + "/bin/sw_vers" }"
  '' else ''
    substituteInPlace unittests/Support/Host.cpp \
      --replace '/usr/bin/sw_vers' "${(builtins.toString darwin.DarwinTools) + "/bin/sw_vers" }"
  '') + optionalString (lib.versionAtLeast release_version "16") ''
```

Should I do that?

Reply:
Okay added it. Let me know if that's good.

Reply:
>  I didn't know why shell sw_vers didn't work in the Makefile so wasn't sure where sw_vers would actually work.

Huh, is it possible `sw_vers` just isn't on the PATH? Maybe you can add it and avoid the need for any patching.

I have no idea if there's a better way, but this looks better to me than hardcoding.

Reply:
`sw_vers` is apparently provided by `darwin` another nixpkg (which is how LLVM gets it). I think the current solution (see updated code) is fine?